text_models/sbert.py

```diff
@@ -13,6 +13,33 @@
 from text_models.bert import BertModelMLM
 
 
+class SbertModelDataset(Dataset):
+    def __init__(self, corpus, disc_ids, n_examples, shuffle=False):
+        if shuffle:
+            data = list(zip(corpus, disc_ids))
+            np.random.shuffle(data)
+            corpus, disc_ids = list(zip(*data))
+
+        self.disc_ids = list(disc_ids)
+        self.corpus = list(corpus)
+        self.n_examples = n_examples
+
+    def __getitem__(self, index):
+        i = 0
+        cur_bound = 0
+        while index > cur_bound:
+            i += 1
+            cur_bound += i + 1
+        j = index - i * (i + 1) // 2
+        i += 1
```

Comment:
You can make it a nested function with a meaningful name (something like `index_to_pair`) and a docstring

Reply:
Added